src/main/java/teammates/common/datatransfer/DataBundle.java

```diff
@@ -45,6 +48,57 @@ public void sanitizeForSaving() {
         sanitizeMapForSaving(profiles);
     }
 
+    public CourseAttributes getCourseById(String courseId) {
+        return this.courses.values()
+                .stream()
+                .filter(course -> course.getId().equals(courseId))
+                .findFirst()
+                .orElseGet(null);
+    }
+
+    public List<StudentAttributes> getStudentsForCourse(String courseId) {
+        return this.students.values()
+                .stream()
+                .filter(student -> student.course.equals(courseId))
+                .collect(Collectors.toList());
+    }
+
+    public List<String> getSectionNamesForCourse(String courseId) {
+        List<String> sections = new ArrayList<>();
+
+        List<StudentAttributes> studentsOfCourse = getStudentsForCourse(courseId);
+        studentsOfCourse.forEach(student -> {
+            if (student.section != null && !sections.contains(student.section)) {
+                sections.add(student.section);
+            }
+        });
+
+        return sections;
+    }
+
+    public List<String> getTeamsForSection(String courseId, String sectionName) {
+        List<String> teams = new ArrayList<>();
+
+        List<StudentAttributes> studentsOfCourse = getStudentsForCourse(courseId);
+        studentsOfCourse.forEach(student -> {
+            if (student.section != null && student.section.equals(sectionName) && student.team != null
+                    && !teams.contains(student.team)) {
+                teams.add(student.team);
+            }
+        });
+
+        return teams;
+    }
+
+    public List<StudentAttributes> getStudentsForTeam(String courseId, String sectionName, String teamName) {
+        return this.students.values()
+                .stream()
+                .filter(student -> student.course.equals(courseId))
+                .filter(student -> student.section.equals(sectionName))
+                .filter(student -> student.team.equals(teamName))
+                .collect(Collectors.toList());
+    }
+
```

Comment:
All of the above methods are used only in test. Let's not add them in production class.

If my observation is correct, you can move all of the above methods to the sole test that uses them.